## 1. Importing Libraries

In [1]:
import os
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split

## 2. Reading Data

In [2]:
PROJECT_DIR = r"C:\Users\Shivay\Desktop\flights-sagemaker-project"
DATA_DIR = "data"

In [3]:
os.path.join(PROJECT_DIR, DATA_DIR)

'C:\\Users\\Shivay\\Desktop\\flights-sagemaker-project\\data'

In [4]:
os.path.join(PROJECT_DIR, DATA_DIR,"flight_price.csv")

'C:\\Users\\Shivay\\Desktop\\flights-sagemaker-project\\data\\flight_price.csv'

In [5]:
def get_data(name):
    file_name = f"{name}.csv"
    file_path = os.path.join(PROJECT_DIR, DATA_DIR, file_name)
    return pd.read_csv(file_path)

In [6]:
flights = get_data("flight_price")
flights

Airline Date_of_Journey    Source Destination  \
0           IndiGo      24/03/2019  Banglore   New Delhi   
1        Air India       1/05/2019   Kolkata    Banglore   
2      Jet Airways       9/06/2019     Delhi      Cochin   
3           IndiGo      12/05/2019   Kolkata    Banglore   
4           IndiGo      01/03/2019  Banglore   New Delhi   
...            ...             ...       ...         ...   
10678     Air Asia       9/04/2019   Kolkata    Banglore   
10679    Air India      27/04/2019   Kolkata    Banglore   
10680  Jet Airways      27/04/2019  Banglore       Delhi   
10681      Vistara      01/03/2019  Banglore   New Delhi   
10682    Air India       9/05/2019     Delhi      Cochin   

                       Route Dep_Time  Arrival_Time Duration Total_Stops  \
0                  BLR → DEL    22:20  01:10 22 Mar   2h 50m    non-stop   
1      CCU → IXR → BBI → BLR    05:50         13:15   7h 25m     2 stops   
2      DEL → LKO → BOM → COK    09:25  04:25 10 Jun      19h     2 stops   
3            CCU → NAG → BLR    18:05         23:30   5h 25m      1 stop   
4            BLR → NAG → DEL    16:50         21:35   4h 45m      1 stop   
...                      ...      ...           ...      ...         ...   
10678              CCU → BLR    19:55         22:25   2h 30m    non-stop   
10679              CCU → BLR    20:45         23:20   2h 35m    non-stop   
10680              BLR → DEL    08:20         11:20       3h    non-stop   
10681              BLR → DEL    11:30         14:10   2h 40m    non-stop   
10682  DEL → GOI → BOM → COK    10:55         19:15   8h 20m     2 stops   

      Additional_Info  Price  
0             No info   3897  
1             No info   7662  
2             No info  13882  
3             No info   6218  
4             No info  13302  
...               ...    ...  
10678         No info   4107  
10679         No info   4145  
10680         No info   7229  
10681         No info  12648  
10682         No info  11753  

[10683 rows x 11 columns]

In [7]:
flights.shape

(10683, 11)

In [8]:
flights.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [9]:
flights.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


- The dataset contains 10,683 rows and 11 features
- Columns `Route` and `Total_Stops` have missing value each
- The data types of some features isn't appropriate

## 3. Preliminary Analysis

### 3.1 Check Data Types

In [10]:
flights.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [11]:
flights.dtypes

Airline            object
Date_of_Journey    object
Source             object
Destination        object
Route              object
Dep_Time           object
Arrival_Time       object
Duration           object
Total_Stops        object
Additional_Info    object
Price               int64
dtype: object

In [12]:
# to check what does 'object mean'
value = flights.Date_of_Journey.iloc[0]
value
# its string

'24/03/2019'

In [13]:
value = flights.Dep_Time.iloc[3]
value

'18:05'

### 3.2 Check for Duplicates

In [14]:
flights.duplicated()

0        False
1        False
2        False
3        False
4        False
         ...  
10678    False
10679    False
10680    False
10681    False
10682    False
Length: 10683, dtype: bool

In [15]:
# duplicate value except first occurence-->220
flights.duplicated().sum()

220

In [16]:
# all duplicate occurences values-->415
(
	flights
	.loc[flights.duplicated(keep=False)]
    .sort_values(["Airline", "Date_of_Journey", "Source", "Destination","Dep_Time"])
)

Airline Date_of_Journey    Source Destination                  Route  \
6321  Air India      01/03/2019  Banglore   New Delhi  BLR → BOM → AMD → DEL   
9848  Air India      01/03/2019  Banglore   New Delhi  BLR → BOM → AMD → DEL   
572   Air India      03/03/2019  Banglore   New Delhi              BLR → DEL   
8168  Air India      03/03/2019  Banglore   New Delhi              BLR → DEL   
1495  Air India       1/04/2019   Kolkata    Banglore  CCU → DEL → COK → BLR   
...         ...             ...       ...         ...                    ...   
2870   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
1535   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
3711   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
2634    Vistara      24/03/2019  Banglore   New Delhi              BLR → DEL   
5470    Vistara      24/03/2019  Banglore   New Delhi              BLR → DEL   

     Dep_Time  Arrival_Time Duration Total_Stops  \
6321    08:50  23:55 02 Mar   39h 5m     2 stops   
9848    08:50  23:55 02 Mar   39h 5m     2 stops   
572     21:10         23:55   2h 45m    non-stop   
8168    21:10         23:55   2h 45m    non-stop   
1495    10:00  01:20 02 Apr  15h 20m     2 stops   
...       ...           ...      ...         ...   
2870    05:45         08:35   2h 50m    non-stop   
1535    20:30         23:20   2h 50m    non-stop   
3711    20:30         23:20   2h 50m    non-stop   
2634    11:30         14:10   2h 40m    non-stop   
5470    11:30         14:10   2h 40m    non-stop   

                   Additional_Info  Price  
6321                       No info  17135  
9848                       No info  17135  
572                        No info   7591  
8168                       No info   7591  
1495                       No info  10408  
...                            ...    ...  
2870  No check-in baggage included   4273  
1535  No check-in baggage included   3873  
3711  No check-in baggage included   3873  
2634                       No info   5403  
5470                       No info   5403  

[415 rows x 11 columns]

### 3.3 observations

- The type of `Date_of_Journey`, `Dep_Time` and `Arrival_Time` should be changed to datetime
- The type of `Duration` and `Total_Stops` is mixed. It should be numeric type.
- There're 220 duplicates. These should be removed.

## 4. Detailed Analysis

### Airline

In [17]:
flights.Airline.unique()

array(['IndiGo', 'Air India', 'Jet Airways', 'SpiceJet',
       'Multiple carriers', 'GoAir', 'Vistara', 'Air Asia',
       'Vistara Premium economy', 'Jet Airways Business',
       'Multiple carriers Premium economy', 'Trujet'], dtype=object)

In [19]:
# method chaining-- made change more readable
(
	flights
	.Airline
	.str.replace(" Premium economy", "")
    .str.replace(" Business", "")
    .str.title() # 'Multiple carriers'-- to fix this inconsistency(caps M small c)
    .unique()
	# .str.title()
)

array(['Indigo', 'Air India', 'Jet Airways', 'Spicejet',
       'Multiple Carriers', 'Goair', 'Vistara', 'Air Asia', 'Trujet'],
      dtype=object)

### Date_of_Journey

In [24]:
flights.Date_of_Journey

0        24/03/2019
1         1/05/2019
2         9/06/2019
3        12/05/2019
4        01/03/2019
            ...    
10678     9/04/2019
10679    27/04/2019
10680    27/04/2019
10681    01/03/2019
10682     9/05/2019
Name: Date_of_Journey, Length: 10683, dtype: object

In [26]:
# pd.to_datetime(flights.Date_of_Journey)  # error
pd.to_datetime(flights.Date_of_Journey,dayfirst = True) 

0       2019-03-24
1       2019-05-01
2       2019-06-09
3       2019-05-12
4       2019-03-01
           ...    
10678   2019-04-09
10679   2019-04-27
10680   2019-04-27
10681   2019-03-01
10682   2019-05-09
Name: Date_of_Journey, Length: 10683, dtype: datetime64[ns]

### Source

In [30]:
flights.Source.unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

### Destination

In [31]:
flights.Destination.unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

### Dept_Time

In [66]:
flights.Dep_Time

0        22:20
1        05:50
2        09:25
3        18:05
4        16:50
         ...  
10678    19:55
10679    20:45
10680    08:20
10681    11:30
10682    10:55
Name: Dep_Time, Length: 10683, dtype: object

In [67]:
# to chech if any Dep_Time value different from 0-9 and :
(
    flights
    .Dep_Time
    .loc[lambda ser:ser.str.contains("[^0-9:]")]
)

Series([], Name: Dep_Time, dtype: object)

In [71]:
pd.to_datetime(flights.Dep_Time)

C:\Users\Shivay\AppData\Local\Temp\ipykernel_20244\2682404337.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  pd.to_datetime(flights.Dep_Time)


0       2024-05-17 22:20:00
1       2024-05-17 05:50:00
2       2024-05-17 09:25:00
3       2024-05-17 18:05:00
4       2024-05-17 16:50:00
                ...        
10678   2024-05-17 19:55:00
10679   2024-05-17 20:45:00
10680   2024-05-17 08:20:00
10681   2024-05-17 11:30:00
10682   2024-05-17 10:55:00
Name: Dep_Time, Length: 10683, dtype: datetime64[ns]

### Arrival_Time

In [52]:
flights.Arrival_Time

0        01:10 22 Mar
1               13:15
2        04:25 10 Jun
3               23:30
4               21:35
             ...     
10678           22:25
10679           23:20
10680           11:20
10681           14:10
10682           19:15
Name: Arrival_Time, Length: 10683, dtype: object

In [56]:
(
    flights
    .Arrival_Time
    .loc[lambda ser:ser.str.contains("[^0-9:]")]
    .str.split(" ", n=1)
	.str.get(1)
	.unique()
)

array(['22 Mar', '10 Jun', '13 Mar', '02 Mar', '10 May', '04 Mar',
       '13 Jun', '28 May', '19 Mar', '07 May', '02 Jun', '16 Jun',
       '19 May', '16 May', '28 Jun', '02 May', '28 Mar', '19 Jun',
       '04 Apr', '25 Mar', '07 Mar', '25 Jun', '07 Jun', '25 May',
       '13 May', '16 Mar', '22 May', '10 Apr', '04 Jun', '20 May',
       '28 Apr', '25 Apr', '10 Mar', '19 Apr', '13 Apr', '02 Apr',
       '23 Mar', '22 Apr', '11 May', '07 Apr', '03 May', '08 Mar',
       '03 Mar', '05 Mar', '22 Jun', '04 May', '26 May', '16 Apr',
       '26 Jun', '29 May', '29 Jun', '29 Mar', '23 May', '17 Jun'],
      dtype=object)

### Duration

In [72]:
flights.Duration

0        2h 50m
1        7h 25m
2           19h
3        5h 25m
4        4h 45m
          ...  
10678    2h 30m
10679    2h 35m
10680        3h
10681    2h 40m
10682    8h 20m
Name: Duration, Length: 10683, dtype: object

In [85]:
# flights only having hours
(
    flights
    .Duration
    .loc[lambda ser:~ser.str.contains("m")]
    # .unique()
)

2        19h
18       23h
33       22h
44       12h
53        3h
        ... 
10591    23h
10638    14h
10639    38h
10673    15h
10680     3h
Name: Duration, Length: 1031, dtype: object

In [84]:
# flights.Duration == '38h'

In [76]:
# flights only having min
# there can be no flight taking 5 mins only
# row having index-6474 have duration of 5 min only
(
    flights
    .Duration
    .loc[lambda ser:~ser.str.contains("h")]
    # .unique()
)

6474    5m
Name: Duration, dtype: object

In [77]:
flights.iloc[[6474]]

Airline Date_of_Journey  Source Destination                  Route  \
6474  Air India       6/03/2019  Mumbai   Hyderabad  BOM → GOI → PNQ → HYD   

     Dep_Time Arrival_Time Duration Total_Stops Additional_Info  Price  
6474    16:50        16:55       5m     2 stops         No info  17327

- The observation indexed 6474 has duration of 5 minutes. This is clearly wrong. Will delete this observation.

In [107]:
(
    flights
    .Duration
    .drop(index = [6474])
    .str.split(" ", expand = True)
    .set_axis(['hour','minute'],axis = 1)
    .assign(
        hour = lambda _df:(
            _df
            .hour
            .str.replace('h', '')
            .astype(int)
            .mul(60)
        ),
        minute = lambda df_:(
            df_
            .minute
            .str.replace('m', '')
            .fillna('0')
            .astype(int)
        )
    )
    .sum(axis=1)
    # .dtypes
    # .isna().sum()
)

0         170
1         445
2        1140
3         325
4         285
         ... 
10678     150
10679     155
10680     180
10681     160
10682     500
Length: 10682, dtype: int64

In [112]:
# to validate duration column
(
    flights
    .Duration
    .drop(index = [6474])
    .str.split(" ", expand = True)
    .set_axis(['hour','minute'],axis = 1)
    .assign(
        hour = lambda _df:(
            _df
            .hour
            .str.replace('h', '')
            .astype(int)
            .mul(60)
        ),
        minute = lambda df_:(
            df_
            .minute
            .str.replace('m', '')
            .fillna('0')
            .astype(int)
        )
    )
    .sum(axis=1)
    # .dtypes
    # .isna().sum()
    .rename('duration_minutes')
    .to_frame()  # to make df
    .join(flights.Duration.drop(index = 6474))
)


duration_minutes Duration
0                   170   2h 50m
1                   445   7h 25m
2                  1140      19h
3                   325   5h 25m
4                   285   4h 45m
...                 ...      ...
10678               150   2h 30m
10679               155   2h 35m
10680               180       3h
10681               160   2h 40m
10682               500   8h 20m

[10682 rows x 2 columns]

### Total_Stops

In [117]:
flights.Total_Stops.unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', nan, '4 stops'],
      dtype=object)

In [125]:
(
    flights
    .Total_Stops
    .str.replace('non-stop','0')
    .str.replace(' stops?', '', regex = True)
    .pipe(lambda ser:pd.to_numeric(ser))
    .unique()
)

array([ 0.,  2.,  1.,  3., nan,  4.])

### Additional_Info

In [131]:
flights.Additional_Info.unique()

array(['No info', 'In-flight meal not included',
       'No check-in baggage included', '1 Short layover', 'No Info',
       '1 Long layover', 'Change airports', 'Business class',
       'Red-eye flight', '2 Long layover'], dtype=object)

In [144]:
# to see actual column names

In [25]:
flights.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

## 5. Cleaning Operations

In [135]:
def convert_to_minutes(ser):
	return (
		ser
		.str.split(" ", expand=True)
		.set_axis(["hour", "minute"], axis=1)
		.assign(
			hour=lambda df_: (
				df_
				.hour
				.str.replace("h", "")
				.astype(int)
				.mul(60)
			),
			minute=lambda df_: (
				df_
				.minute
				.str.replace("m", "")
				.fillna("0")
				.astype(int)
			)
		)
		.sum(axis=1)
	)


def clean_data(df):
	return (
		df
        .drop(index = [6474])
        .drop_duplicates()
        .assign(**{
			col: df[col].str.strip() 
			for col in df.select_dtypes(include="O").columns  # "o" -means object only
		})
        .rename(columns=str.lower)
        .assign(
            airline=lambda df_:(
                df_
                .airline
                .str.replace(" Premium economy", "")
                .str.replace(" Business", "")
                .str.title()
            ),
            date_of_journey=lambda df_: pd.to_datetime(df_.date_of_journey, dayfirst=True),
            dep_time=lambda df_: pd.to_datetime(df_.dep_time).dt.time,
            arrival_time=lambda df_: pd.to_datetime(df_.arrival_time).dt.time,
            duration=lambda df_: df_.duration.pipe(convert_to_minutes),
            total_stops = lambda df_:(
                df_
                .total_stops
                .str.replace('non-stop','0')
                .str.replace(' stops?', '', regex = True)
                .pipe(lambda ser:pd.to_numeric(ser))
            ),
            additional_info=lambda df_: df_.additional_info.replace("No info", "No Info")
        )
        .drop(columns = "route")
    )

In [136]:
flights_cleaned = clean_data(flights)
flights_cleaned

C:\Users\Shivay\AppData\Local\Temp\ipykernel_20244\119664678.py:45: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  dep_time=lambda df_: pd.to_datetime(df_.dep_time).dt.time,
C:\Users\Shivay\AppData\Local\Temp\ipykernel_20244\119664678.py:46: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  arrival_time=lambda df_: pd.to_datetime(df_.arrival_time).dt.time,


airline date_of_journey    source destination  dep_time  \
0           Indigo      2019-03-24  Banglore   New Delhi  22:20:00   
1        Air India      2019-05-01   Kolkata    Banglore  05:50:00   
2      Jet Airways      2019-06-09     Delhi      Cochin  09:25:00   
3           Indigo      2019-05-12   Kolkata    Banglore  18:05:00   
4           Indigo      2019-03-01  Banglore   New Delhi  16:50:00   
...            ...             ...       ...         ...       ...   
10678     Air Asia      2019-04-09   Kolkata    Banglore  19:55:00   
10679    Air India      2019-04-27   Kolkata    Banglore  20:45:00   
10680  Jet Airways      2019-04-27  Banglore       Delhi  08:20:00   
10681      Vistara      2019-03-01  Banglore   New Delhi  11:30:00   
10682    Air India      2019-05-09     Delhi      Cochin  10:55:00   

      arrival_time  duration  total_stops additional_info  price  
0         01:10:00       170          0.0         No Info   3897  
1         13:15:00       445          2.0         No Info   7662  
2         04:25:00      1140          2.0         No Info  13882  
3         23:30:00       325          1.0         No Info   6218  
4         21:35:00       285          1.0         No Info  13302  
...            ...       ...          ...             ...    ...  
10678     22:25:00       150          0.0         No Info   4107  
10679     23:20:00       155          0.0         No Info   4145  
10680     11:20:00       180          0.0         No Info   7229  
10681     14:10:00       160          0.0         No Info  12648  
10682     19:15:00       500          2.0         No Info  11753  

[10462 rows x 10 columns]

In [23]:
# check if changes done
clean_data(flights).airline.unique()

array(['Indigo', 'Air India', 'Jet Airways', 'Spicejet',
       'Multiple Carriers', 'Goair', 'Vistara', 'Air Asia', 'Trujet'],
      dtype=object)

In [28]:
# to check changes in date_of_journey
clean_data(flights).dtypes  # date_of_journey data type is-->datetime64[ns]

airline                    object
date_of_journey    datetime64[ns]
source                     object
destination                object
route                      object
dep_time                   object
arrival_time               object
duration                   object
total_stops                object
additional_info            object
price                       int64
dtype: object

### 6. Split the data

In [145]:
flights_final = flights_cleaned.sample(2500)

In [146]:
# flights_final = flights_cleaned.copy()

In [147]:
X = flights_final.drop(columns="price")
y = flights_final.price.copy()

In [148]:
X_, X_test, y_, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
X_train, X_val, y_train, y_val = train_test_split(X_, y_, test_size=0.2, random_state=42)

print(X_train.shape, y_train.shape)
print(X_val.shape, y_val.shape)
print(X_test.shape, y_test.shape)

(1600, 9) (1600,)
(400, 9) (400,)
(500, 9) (500,)


## 7. Export the Subsets

In [149]:
def export_data(X, y, name):
	file_name = f"{name}.csv"
	file_path = os.path.join(PROJECT_DIR, DATA_DIR, file_name)

	X.join(y).to_csv(file_path, index=False)

	return pd.read_csv(file_path).head()

In [150]:
export_data(X_train, y_train, "train")

airline date_of_journey    source destination  dep_time  \
0        Jet Airways      2019-05-15   Kolkata    Banglore  09:35:00   
1        Jet Airways      2019-06-21  Banglore       Delhi  17:45:00   
2        Jet Airways      2019-03-09     Delhi      Cochin  06:50:00   
3  Multiple Carriers      2019-06-27     Delhi      Cochin  08:45:00   
4             Indigo      2019-06-24     Delhi      Cochin  14:20:00   

  arrival_time  duration  total_stops              additional_info  price  
0     23:35:00       840          1.0                      No Info  14151  
1     20:45:00       180          0.0  In-flight meal not included   7754  
2     04:25:00      1295          1.0                      No Info  16289  
3     19:15:00       630          1.0                      No Info   8266  
4     17:35:00       195          0.0                      No Info   5000

In [151]:
export_data(X_val, y_val, "val")

airline date_of_journey   source destination  dep_time arrival_time  \
0     Indigo      2019-03-03  Kolkata    Banglore  10:10:00     12:55:00   
1  Air India      2019-04-01  Kolkata    Banglore  14:15:00     16:45:00   
2     Indigo      2019-05-21  Chennai     Kolkata  07:55:00     10:15:00   
3  Air India      2019-06-15   Mumbai   Hyderabad  19:30:00     20:45:00   
4    Vistara      2019-03-06   Mumbai   Hyderabad  10:25:00     22:55:00   

   duration  total_stops additional_info  price  
0       165          0.0         No Info   6565  
1       150          0.0         No Info   6245  
2       140          0.0         No Info   3850  
3        75          0.0         No Info   3100  
4       750          1.0         No Info  12395

In [152]:
export_data(X_test, y_test, "test")

airline date_of_journey    source destination  dep_time  \
0  Multiple Carriers      2019-06-12     Delhi      Cochin  14:00:00   
1        Jet Airways      2019-05-12   Kolkata    Banglore  20:25:00   
2          Air India      2019-04-01   Kolkata    Banglore  16:50:00   
3          Air India      2019-06-12  Banglore       Delhi  21:05:00   
4              Goair      2019-06-06  Banglore       Delhi  11:40:00   

  arrival_time  duration  total_stops              additional_info  price  
0     01:30:00       690          1.0                      No Info  13377  
1     21:05:00      1480          1.0  In-flight meal not included  10703  
2     18:30:00      1540          1.0                      No Info   7893  
3     23:55:00       170          0.0                      No Info   5228  
4     14:30:00       170          0.0                      No Info   3898